src/api/class-request.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * WP Video Sync: API request integration.
+ *
+ * @package wp-video-sync
+ */
+
+namespace Alley\WP\WP_Video_Sync\API;
+
+use Alley\WP\WP_Video_Sync\Interfaces\API_Requester;
+
+/**
+ * Perform an API request.
+ */
+class Request {
```

Comment:
This is interesting - I've always implemented API adapters as self-contained, rather than passing them as dependencies using DI into a generic request class. Is this a pattern that we're using elsewhere? I can see how it would cut down on repeated code, but I can also see how it would introduce some confusion into utilization (you have to set the properties of the API object including request URL, then pass it to the request class for execution, versus calling a method on the API class to get the data). A way to make this a bit more user friendly would be to encapsulate this functionality in the JW Player API class like so:
```php
public function get_videos_after( DateTimeImmutable $updated_after, int $batch_size ): array {
	$this->set_request_url(
		$updated_after->format( 'Y-m-d' ),
		$batch_size
	);

	return ( new Request( $this ) )->get();
}
```
That code could be moved out of the adapter and into the API wrapper, and then the adapter would call `$this->jw_player_api->get_videos_after( $updated_after, $batch_size )` and would otherwise proceed as normal. That makes the way the API class makes its requests an implementation detail, rather than something that the adapter class needs to worry about.

Reply:
This has been updated in the latest commit: https://github.com/alleyinteractive/wp-video-sync/pull/10/commits/739708210a76ba8c0dddf83f979439786f108772